src/main/java/functions/TaskList.java

```diff
@@ -0,0 +1,92 @@
+package functions;
+
+import datatypes.Deadline;
+import datatypes.Event;
+import datatypes.Task;
+import datatypes.Todo;
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> arrayOfTasks = new ArrayList<>();
+    private int numberOfTasks;
+
+    public TaskList() {
+        setNumberOfTasks(0);
+    }
+
+    public int getNumberOfTasks() {
+        return numberOfTasks;
+    }
+
+    public void setNumberOfTasks(int numberOfTasks) {
+        this.numberOfTasks=numberOfTasks;
+    }
+
+    public void printAllTasks() {
+        if(arrayOfTasks.isEmpty()) {
+            System.out.println("No tasks added yet!");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i<arrayOfTasks.size();i++){
+                Task currentTask = arrayOfTasks.get(i);
+                System.out.println(i+1 + "." + currentTask.toString());
+            }
+        }
+    }
+
+    public void addTask(AnalyseCommand usercommands) {
+
+        Task newTask;
+
+        switch (usercommands.getCommandType()) {
+        case "todo":
+            newTask= new Todo(usercommands.getCommandDescription());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks()+1);
+            printAddTaskMessage(newTask);
+            break;
+        case "deadline":
+            newTask= new Deadline(usercommands.getCommandDescription(), usercommands.getCommandTime());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks()+1);
+            printAddTaskMessage(newTask);
+            break;
+        case "event":
+            newTask= new Event(usercommands.getCommandDescription(), usercommands.getCommandTime());
+            arrayOfTasks.add(newTask);
+            setNumberOfTasks(getNumberOfTasks()+1);
+            printAddTaskMessage(newTask);
```

Comment:
A few common lines of code can be refactored out to reduce duplication. The lines are :
arrayOfTasks.add(newTask);
setNumberOfTasks(getNumberOfTasks()+1);
printAddTaskMessage(newTask);
Perhaps they could go in a method and call that method instead?